lib/Button/index.js

```diff
@@ -33,42 +36,31 @@ class Button extends Component {
     };
 
     this.handleOnClick = this.handleOnClick.bind(this);
+    this.getTypeClasses = this.getTypeClasses.bind(this);
   }
 
   handleOnClick() {
     this.setState({ disabled: this.props.disableOnClick });
     this.props.clickHandler();
   }
 
+  getTypeClasses() {
```

Comment:
I thought it'd be a neat idea to move the previous "type" prop for a "types" prop which allows us to define multiple types that are then used to string together the modifier classes e.g. `primary` and `collapsed`.

I wasn't super keen on the switch statement used in the previous component. Mostly because when I wanted to piece together types like `clear` and `collapsed` I had to add a brand new type.

However, the previous version did do a great job at describing the main types of Button you could use. What do you think @magalhini?